gnat2goto/driver/tree_walk.adb

```diff
@@ -2343,63 +2343,219 @@ package body Tree_Walk is
       end if;
       N_Orig := Original_Node (N);
       case Pragma_Name (N_Orig) is
-         when Name_Assert | Name_Assume | Name_Loop_Invariant =>
+         when Name_Assert |
+              Name_Assume |
+              Name_Precondition |
+              Name_Assert_And_Cut |
+            --  Assert and introduce a cut point: the prover can safely forget
+            --  evaluations of local variables and only assume the asserted
+            --  condition. This could be used in symex (making it concolic)
+            --  but is only an optimization.
+            Name_Loop_Invariant =>
+            --  Equivalent to assert but also introduces a cut point wrt. the
+            --  variables local to the loop.
             Do_Pragma_Assert_or_Assume (N_Orig, Block);
-         when Name_Annotate =>
-            --  Ignore here. Rather look for those when we process a node.
-            null;
-         when Name_Refined_State | Name_Refined_Global |
+         when Name_Refined_State |
+              Name_Refined_Global |
               Name_Refined_Depends =>
+            --  We are effectively ignoring the refinement at this point
+            --  Using it would (probably) require modification to CBMC
```

Comment:
Really?  Unless the pragma is used for something else, the refinement things that old-SPARK used to do weren't that commonly used.

Reply:
Well at least some of the repositories we're considering to for integration testing that use Spark use refining, but maybe not all of them.

Reply:
Thanks for the head's up; let's solve that problem when we come to it.  My understanding was that pragma refine* was a SPARK 2014 construct, rather than an old-SPARK construct and I think much of the code we have planned for integration testing is old-SPARK.